daffodil-cli/src/main/scala/org/apache/daffodil/Main.scala

```diff
@@ -1236,8 +1266,77 @@ object Main {
         }
       }
 
-      // Required to avoid "match may not be exhaustive", but should never happen
-      case _ => Assert.impossible()
+      case Some(conf.exi) => {
+        var rc = ExitCode.Success
+        val exiOpts = conf.exi
+        val channel = exiOpts.output.toOption match {
+          case Some("-") | None => Channels.newChannel(STDOUT)
+          case Some(file) => new FileOutputStream(file).getChannel()
+        }
+        val output = Channels.newOutputStream(channel)
+
+        val inputStream = exiOpts.infile.toOption match {
+          case Some("-") | None => STDIN
+          case Some(file) => {
+            val f = new File(file)
+            new FileInputStream(f)
+          }
+        }
+        val input = new InputSource(inputStream)
+
+        val exiFactory = DefaultEXIFactory.newInstance
+        if (exiOpts.schema.isDefined) try {
+          val grammarFactory = GrammarFactory.newInstance
+          val grammar = grammarFactory.createGrammars(exiOpts.schema.toOption.get.toString, DFDLCatalogResolver.get)
+          exiFactory.setGrammars(grammar)
+        } catch {
+          case e: EXIException => {
+            Logger.log.error(s"Error creating EXI grammar for the supplied schema: ${ Misc.getSomeMessage(e).get }")
+            rc = ExitCode.Failure
+          }
+        }
+
+        (exiOpts.decode.toOption.get, rc) match {
+          case (true, ExitCode.Success) => { // Decoding
+            val exiSource = new EXISource(exiFactory)
+            exiSource.setInputSource(input)
+
+            val result = new StreamResult(output)
+            val tf = TransformerFactory.newInstance()
+            val transformer = tf.newTransformer
+            try {
+              transformer.transform(exiSource, result)
+            } catch {
+              case e: Exception => {
```

Comment:
Can you add a comment why we catch a generic Exception? We usually try to avoid that except in cases where there are no other reasonable exceptions to catch, in which case it's worth a comment.